test/sol5/kyberProxyIntegration.js

```diff
@@ -0,0 +1,710 @@
+let defaultRebateBps = 2000; // 20%
+
+let initVictorStake = mulPrecision(2000);
+let initMikeStake = mulPrecision(2000);
+
+// Staking data
+let daoSetter;
+let currentBlock;
+let epochPeriod = 20;
+let startBlock;
+let stakingContract;
+
+//fee hanlder related
+let KNC;
+let burnBlockInterval = new BN(30);
+
+//reserve data
+//////////////
+let reserveInstances = [];
+
+//tokens data
+////////////
+let numTokens = 5;
+let tokens = [];
+let tokenDecimals = [];
+let srcToken;
+let destToken;
+let srcDecimals;
+let ethSrcQty;
+let srcQty;
+
+contract('Proxy + Network + MatchingEngine + FeeHandler + Staking + DAO integrations', function(accounts) {
+    before("one time global init", async() => {
+        //init accounts
+        operator = accounts[1];
+        alerter = accounts[2];
+        taker = accounts[3];
+        admin = accounts[5]; // we don't want admin as account 0.
+        campCreator = accounts[6];
+        daoSetter = accounts[7];
+        mike = accounts[8];
+        victor = accounts[9];
+
+        //init KNC
+        KNC = await TestToken.new("kyber network crystal", "KNC", 18);
+
+        //deploy network
+        network = await KyberNetwork.new(admin);
+
+        // init proxy
+        networkProxy = await KyberNetworkProxy.new(admin);
+
+        // FeeHandler init
+        feeHandler = await FeeHandler.new(daoSetter, networkProxy.address, network.address, KNC.address, burnBlockInterval);
+        // Staking & DAO init
+        currentBlock = await Helper.getCurrentBlock();
+        await deployContracts(40, currentBlock + 350, 10);
+        await setupSimpleStakingData();
+
+        // set DAO for feeHandler
+        await feeHandler.setDaoContract(daoContract.address, {from: daoSetter});
+
+        //init matchingEngine
+        matchingEngine = await MatchingEngine.new(admin);
+        await matchingEngine.setNetworkContract(network.address, {from: admin});
+        await matchingEngine.setFeePayingPerReserveType(true, true, true, false, true, true, {from: admin});
+
+        rateHelper = await RateHelper.new(admin);
+        await rateHelper.setContracts(matchingEngine.address, daoContract.address, {from: admin});
+
+        // setup proxy
+        await networkProxy.setKyberNetwork(network.address, {from: admin});
+        await networkProxy.setHintHandler(matchingEngine.address, {from: admin});
+
+        //init tokens
+        for (let i = 0; i < numTokens; i++) {
+            tokenDecimals[i] = new BN(15).add(new BN(i));
+            token = await TestToken.new("test" + i, "tst" + i, tokenDecimals[i]);
+            tokens[i] = token;
+        }
+
+        srcToken = tokens[0];
+        srcDecimals = tokenDecimals[0];
+        srcQty = new BN(10).mul((new BN(10)).pow(new BN(srcDecimals)));
+        destToken = tokens[1];
+        ethSrcQty = precisionUnits;
+
+        // init and setup reserves
+        let result = await nwHelper.setupReserves(network, tokens, 1, 2, 3, 0, accounts, admin, operator);
```

Comment:
3rd type not supported yet. so the '3' has no effect.

Reply:
Removed the third type